code/API_definitions/number_verification.yaml

```diff
@@ -70,8 +70,8 @@ paths:
       summary: Verifies if the received hashed/plain text phone number matches the phone number associated with the access token
       description: |
         Verifies if the specified phone number (either in plain text or hashed format) matches the one that the user is currently using. Only one of the plain or hashed formats must be provided.
-         - The number verification will be done for the user that has authenticated via mobile network and so their `sub` is in the access token
-         - It returns true/false depending on if the hashed phone number received as input matches the authenticated user's `device phone number` associated to the access token
+        - The number verification will be done for the user that has authenticated via mobile network and so their `sub` is in the access token
```

Comment:
if we use "OpenID Connect" sub is one of id_token claims.

@fernandopradocabrillo please review

Reply:
afaik access_tokens also contains the `sub`: https://auth0.com/docs/authenticate/login/oidc-conformant-authentication/oidc-adoption-access-tokens#standard-claims

Reply:
it depends...

following RFC:
```
Access tokens can have different formats, structures, and methods of
   utilization (e.g., cryptographic properties) based on the resource
   server security requirements.  Access token attributes and the
   methods used to access protected resources are beyond the scope of
   this specification and are defined by companion specifications such
   as RFC6750.
```
[RFC6750](https://datatracker.ietf.org/doc/html/rfc6750)
[Access Token](https://datatracker.ietf.org/doc/html/rfc6749#section-1.4)

and sub required claim inside ID Token, that is something specific for OpenId Connect, as it is designed for authentication (& authorization).
[ID Token](https://openid.net/specs/openid-connect-core-1_0.html#IDToken)

So you can state that:

sub is **build in OpenID Connect**, as it is a part of id_token, and is **optional for OAUTH2.0**  as it depends on _resource server security requirements_.

Therefore, if we're willing to use _access_tokens_ we should use OAUTH2.0 security scheme.